platforms/metagram/src/lib/fragments/Comment/Comment.svelte

```diff
@@ -0,0 +1,147 @@
+<script lang="ts">
+	import { Like } from '$lib/icons';
+	import { Avatar } from '$lib/ui';
+	import { cn } from '$lib/utils';
+	import type { HTMLAttributes } from 'svelte/elements';
+	import type { CommentType } from '$lib/types';
+
+	interface ICommentProps extends HTMLAttributes<HTMLElement> {
+		comment: CommentType;
+		handleReply: () => void;
+	}
+
+	let visibleReplies = $state(2);
+
+	const showMoreReplies = () => {
+		visibleReplies = comment.replies.length;
+	};
+
+	let { comment, handleReply, ...restProps }: ICommentProps = $props();
+</script>
+
+<article {...restProps} class={cn([restProps.class].join(' '))}>
+	<div class="align-start flex gap-2">
+		<Avatar src={comment.userImgSrc} size="sm" />
+		<div>
+			<h3 class="font-semibold text-black">{comment.name}</h3>
+			<p class="text-black-600 mt-0.5">{comment.comment}</p>
+		</div>
+	</div>
+	<div class="ms-12 mt-2 flex items-center gap-2">
+		<button
+			onclick={() => {
+				if (!comment.isUpVoted) {
+					comment.upVotes++;
+					comment.isUpVoted = true;
+					comment.isDownVoted = false;
```

Comment:
This should be combined into a "voteStatus" variable like this 

type VoteStatus = "up" | "down" | "none";

interface Comment {
	upVotes: number;
	downVotes: number;
	voteStatus: VoteStatus;
}

To avoid inconsistent states.

```suggestion
if (comment.voteStatus !== "up") {
		// Remove previous downvote if it exists
		if (comment.voteStatus === "down") {
			comment.downVotes--;
		}

		comment.upVotes++;
		comment.voteStatus = "up";
```